### PR TITLE
*: use origin-v4.0:kube-client-agent

### PR DESCRIPTION
**Pull request by hexfusion**
This PR adds the newly built kube-client-agent image registry.svc.ci.openshift.org/openshift/origin-v4.0:kube-client-agent replacing the coreos/kubecsr upstream.

Because of chicken and egg I need to set the default to pass tests and will update this after the flag is added to installer.

### CODE REVIEW

**Comment by hexfusion** [comment]
updating test yamls

**Comment by kikisdeliveryservice** [comment]
hi @hexfusion !

Could you update the commit message to include (in the body):  `after related flag PR merged into installer, uncomment bootstrapCmd.MarkFlagRequired("kube-client-agent-image") in bootstrap.go`

**Comment by hexfusion** [comment]
@kikisdeliveryservice sure thing all set.

**Comment by kikisdeliveryservice** [comment]
thanks @hexfusion !!

**Comment by hexfusion** [comment]
removing default for testing release.

**Comment by cgwalters** [comment]
Oh man this again...luckily the scars are fresh from the last time:
https://github.com/openshift/machine-config-operator/pull/471

It looks like you probably saw https://github.com/openshift/machine-config-operator/pull/471#issuecomment-467154348 at least?

**Comment by hexfusion** [comment]
@cgwalters @abhinavdahiya  with the above PR and installer with https://github.com/openshift/installer/pull/1401 it works can we force 1401? then this?

To confirm my test actions I did the following which was the latest ci release on this PR. My installer as noted contains https://github.com/openshift/installer/pull/1401
```
OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=registry.svc.ci.openshift.org/ci-op-gmkb8pn6/release:latest openshift-install-dev --log-level debug --dir aws-cluster/cluster-dev create cluster
```

**Comment by abhinavdahiya** [comment]
> @cgwalters @abhinavdahiya with the above PR and installer with [openshift/installer#1401](https://github.com/openshift/installer/pull/1401) it works can we force 1401? then this?
> 
> To confirm my test actions I did the following which was the latest ci release on this PR. My installer as noted contains [openshift/installer#1401](https://github.com/openshift/installer/pull/1401)
> 
> ```
> OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=registry.svc.ci.openshift.org/ci-op-gmkb8pn6/release:latest openshift-install-dev --log-level debug --dir aws-cluster/cluster-dev create cluster
> ```

The way you can proceed without force is:

1. in This PR
- hard code the image to registry one [here](https://github.com/openshift/machine-config-operator/pull/538/files#diff-d6f38878757202a810cfa8038e7ae122R90), do not use the one set by flag.
- hard code the image to registry one [here](https://github.com/openshift/machine-config-operator/pull/538/files#diff-554de5523753fda8c93e7008c9bd947fR329), do not use the one set in config map. 

2. This allows installer PR https://github.com/openshift/installer/pull/1401 to set set the flag,

3. come and switch [here](https://github.com/openshift/machine-config-operator/pull/538/files#diff-d6f38878757202a810cfa8038e7ae122R90) and [here](https://github.com/openshift/machine-config-operator/pull/538/files#diff-554de5523753fda8c93e7008c9bd947fR329) to use the correct images from flag and config map. And also set the flag as required.

**Comment by hexfusion** [comment]
Ci failure unrelated I just ran OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=registry.svc.ci.openshift.org/ci-op-0xsz8z88/release:latest with success.

```
level=error msg="Error: Error applying plan:"
level=error
level=error msg="4 errors occurred:"
level=error msg="\t* module.vpc.aws_route_table_association.route_net[1]: 1 error occurred:"
level=error msg="\t* aws_route_table_association.route_net.1: timeout while waiting for state to become 'success' (timeout: 5m0s)"
level=error
level=error
level=error msg="\t* module.vpc.aws_route_table_association.route_net[0]: 1 error occurred:"
level=error msg="\t* aws_route_table_association.route_net.0: timeout while waiting for state to become 'success' (timeout: 5m0s)"
level=error
level=error
level=error msg="\t* module.vpc.aws_route_table_association.route_net[3]: 1 error occurred:"
level=error msg="\t* aws_route_table_association.route_net.3: timeout while waiting for state to become 'success' (timeout: 5m0s)"
level=error
level=error
level=error msg="\t* module.vpc.aws_route_table_association.route_net[4]: 1 error occurred:"
level=error msg="\t* aws_route_table_association.route_net.4: timeout while waiting for state to become 'success' (timeout: 5m0s)"
level=error
```

**Comment by hexfusion** [comment]
@abhinavdahiya PTAL this will pass if we can leap over resource limit.

File: cmd/machine-config-operator/bootstrap.go [hunk 1]
```diff
@@ -24,22 +24,23 @@ var (
...
+		networkConfigFile    string
+		imagesConfigMapFile  string
+		mccImage             string
+		mcsImage             string
+		mcdImage             string
+		etcdImage            string
+		setupEtcdEnvImage    string
+		infraImage           string
+		destinationDir       string
+		kubeClientAgentImage string
```
Comment (abhinavdahiya) [inline cmd/machine-config-operator/bootstrap.go]: nit: group images together.

File: cmd/machine-config-operator/bootstrap.go [hunk 2]
```diff
@@ -86,6 +93,7 @@ func runBootstrapCmd(cmd *cobra.Command, args []string) {
 		Etcd:                    bootstrapOpts.etcdImage,
 		SetupEtcdEnv:            bootstrapOpts.setupEtcdEnvImage,
 		InfraImage:              bootstrapOpts.infraImage,
+		KubeClientAgentImage:    bootstrapOpts.kubeClientAgentImage,
```
Comment (abhinavdahiya) [inline cmd/machine-config-operator/bootstrap.go]: nit: the struct is already `Images` `KubeClientAgentImage` the Image suffix is redundant.

**Comment by abhinavdahiya** [comment]
> @abhinavdahiya PTAL this will pass if we can leap over resource limit.

LGTM

File: cmd/machine-config-operator/bootstrap.go (hunk 2 above)
Comment (hexfusion) [reply cmd/machine-config-operator/bootstrap.go]: updated

File: cmd/machine-config-operator/bootstrap.go (hunk 1 above)
Comment (hexfusion) [reply cmd/machine-config-operator/bootstrap.go]: sure, updated.

**Pull request merged**